phpBB/phpbb/session.php

```diff
@@ -1599,4 +1554,48 @@ public function update_session($session_data, $session_id = null)
 		$vars = array('session_data', 'session_id');
 		extract($phpbb_dispatcher->trigger_event('core.update_session_after', compact($vars)));
 	}
+
+	public function update_session_infos()
+	{
+		global $db, $request;
+
+		// No need to update if it's a new session. Informations are already inserted by session_create()
+		if (isset($this->data['session_created']) && $this->data['session_created'])
+		{
+			return;
+		}
+
+		// Only update session DB a minute or so after last update or if page changes
+		if ($this->time_now - $this->data['session_time'] > 60 || ($this->update_session_page && $this->data['session_page'] != $this->page['page']))
+		{
+			$sql_ary = array('session_time' => $this->time_now);
+
+			// Do not update the session page for ajax requests, so the view online still works as intended
+			if ($this->update_session_page && !$request->is_ajax())
+			{
+				$sql_ary['session_page'] = substr($this->page['page'], 0, 199);
+				$sql_ary['session_forum_id'] = $this->page['forum'];
+			}
+
+			$db->sql_return_on_error(true);
+
+			$this->update_session($sql_ary);
+
+			$db->sql_return_on_error(false);
+
+			// If the database is not yet updated, there will be an error due to the session_forum_id
+			// @todo REMOVE for 3.0.2
+			if ($result === false)
+			{
+				unset($sql_ary['session_forum_id']);
+
+				$this->update_session($sql_ary);
+			}
```

Comment:
`$result` doesn't exists and fails tests. I will remove this code. So, if this PR is merged, #4007 will be obselete.